doc/HelpInterface.md

```diff
@@ -0,0 +1,89 @@
+# Halibot Help Text Interface
+
+Help text is crucial for new users to learn how to operate modules and Halibot itself.
+While Halibot core does not require module developers to supply help text, it is greatly encouraged to do so.
+This document will define the help text ecosystem that is strongly recommended to adhere to.
+
+# Sample Usage
+
+## How to get help text?
+
+### Via an agent (or "By chatting with it")
+
+Halibot will provide a "Help" module as a default that needs to be enabled.
+This handles the "!help" command, which has the following usage:
+
+```
+someuser> !help
+halibot1> Your friendly bot! Use !help <module> to get specific module help
+halibot1> Loaded modules: admin, help, quote
```

Comment:
What should be displayed here for loaded modules? In the case of multiple instances of the same kind of module, it might be more useful to display the name, and the type of module. Something like `name (class)`?
e.g.
```
> Loaded modules: admin (Admin), help (Help), quote (Quote), magic8 (Quote)
```
In this case, both `quote` and `magic8` are both `Quote` modules.

Alternatively, the class name is only displayed with a flag? `!help -v`?